### PR TITLE
Award new years resolution badge when iteration is created, not solution

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: app/commands/iteration/create.rb [hunk 1]
```diff
@@ -25,6 +25,7 @@ def call
       AwardBadgeJob.perform_later(user, :die_unendliche_geschichte, context: iteration)
       AwardBadgeJob.perform_later(user, :growth_mindset)
       record_activity!(iteration)
+      award_badge!(iteration)
```
Comment (iHiD) [inline app/commands/iteration/create.rb]: Look two lines above - move those into here and pluralise the method name?

File: app/models/badges/new_years_resolution_badge.rb [hunk 2]
```diff
@@ -5,13 +5,11 @@ class NewYearsResolutionBadge < Badge
       'new-years-resolution',
       'Submitted a solution on January 1st'
```
Comment (iHiD) [inline app/models/badges/new_years_resolution_badge.rb]: `Submitted an iteration on January 1st`

**Review by iHiD — APPROVED** [review]
LGTM

File: app/models/badges/new_years_resolution_badge.rb [hunk 3]
```diff
@@ -3,15 +3,15 @@ class NewYearsResolutionBadge < Badge
     seed "New Year's resolution",
       :rare,
       'new-years-resolution',
-      'Submitted a solution on January 1st'
+      'Submitted an iteration on January 1st'
 
-    def self.worth_queuing?(solution:)
-      solution.created_at.yday == 1
+    def self.worth_queuing?(iteration:)
+      [1, 2, 365, 366].include?(iteration.created_at.yday)
```
Comment (ErikSchierboom) [inline app/models/badges/new_years_resolution_badge.rb:9]: 366 is for leap years

**Review by iHiD — APPROVED** [review]

**Pull request merged**